### PR TITLE
Adding generic setup configuration to the README.md

### PR DESCRIPTION
**Pull request by markyjackson-taulia**
Signed-off-by: markyjackson-taulia <marky.r.jackson@gmail.com>

In this PR, I have added a small section with some generic setup configurations. I opened this jira https://issues.jenkins-ci.org/browse/JENKINS-61170 and the corresponding gitter chat that prompted me to do this is here: https://gitter.im/jenkinsci/jenkins?at=5e4ee2c50c451412667bacf5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,30 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
 
 It is not required to run the Jenkins master inside Kubernetes. 
+# Generic Setup
+### Prerequisites
+* A running Kubernetes cluster
+* An instances of Jenkins installed
+* The Jenkins Kubernetes plugin installed
+
+Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 
+“Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’
```
Comment (jglick) [inline README.md]: Unless Jenkins is running in Kubernetes in which case the defaults work.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,30 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
 It is not required to run the Jenkins master inside Kubernetes. 
+# Generic Setup
+### Prerequisites
+* A running Kubernetes cluster
+* An instances of Jenkins installed
+* The Jenkins Kubernetes plugin installed
+
+Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 
+“Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’ 
+appropriately. To test this connection is successful you can use the `Test Connection` button to ensure there is
```
Comment (jglick) [inline README.md]:
```suggestion
appropriately. To test this connection is successful you can use the **Test Connection** button to ensure there is 
```

File: README.md [hunk 3]
```diff
@@ -25,6 +25,30 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
+* The Jenkins Kubernetes plugin installed
+
+Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 
+“Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’ 
+appropriately. To test this connection is successful you can use the `Test Connection` button to ensure there is 
+adequate communication from Jenkins to the Kubernetes cluster, as seen below
+
+![image](Screen%20Shot%202020-02-20%20at%202.00.28%20PM.png)
+
+In addition to that, in the ‘Kubernetes Pod Template’ section, we need to configure the image that will be used to
```
Comment (jglick) [inline README.md]:
```suggestion
In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
```

File: README.md [hunk 4]
```diff
@@ -25,6 +25,30 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
+appropriately. To test this connection is successful you can use the `Test Connection` button to ensure there is 
+adequate communication from Jenkins to the Kubernetes cluster, as seen below
+
+![image](Screen%20Shot%202020-02-20%20at%202.00.28%20PM.png)
+
+In addition to that, in the ‘Kubernetes Pod Template’ section, we need to configure the image that will be used to 
+spin up the agent pod. If you have some custom requirements for your agent, you can build one more dockerfile with the 
+appropriate changes the same way we did for the Jenkins control plane. On the other hand, if you don’t have unique 
+requirements 
+for your agent, you can use the default Jenkins agent image available on the official Docker hub repository. In the
```
Comment (jglick) [inline README.md]: This is all pretty off base I think.

First of all, we do not recommend overriding the `jnlp` container except under unusual circumstances. You pretty much always just use the default one. For actual build steps, you use separate containers, which can be anything, and the `container` step.

Second, it is generally discouraged to even use the global pod template configuration. Better to use the `podTemplate` step and define your requirements directly inside the job—or via some library etc.

The existing samples, for example in #707, show these things in action. No global configuration or custom images are required.

Comment (markyjackson-taulia) [reply README.md]: Okay, I will remove that in favor of the base `jnlp` container.
As for the `podTemplate`, would it make sense to give the user (in terms of steps) both options?

File: README.md (hunk 1 above)
Comment (markyjackson-taulia) [reply README.md]: I will add that point

File: README.md (hunk 4 above)
Comment (jglick) [reply README.md]: I guess. The main reason to use the global pod template definition is to migrate a huge corpus of existing projects (incl. freestyle) to run on Kubernetes without changing job definitions. New users setting up new Kubernetes builds should use the `podTemplate` step as shown in the example snippets.

Comment (markyjackson-taulia) [reply README.md]: okay, I will add that line of wording.
Thank you for helping

File: README.md [hunk 5]
```diff
@@ -25,6 +25,32 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
 see the [Docker image source code](https://github.com/jenkinsci/docker-jnlp-slave).
 
 It is not required to run the Jenkins master inside Kubernetes. 
+# Generic Setup
+### Prerequisites
+* A running Kubernetes cluster
+* An instances of Jenkins installed
```
Comment (Vlatombe) [inline README.md]:
```suggestion
* A Jenkins instance installed
```

File: README.md [hunk 6]
```diff
@@ -25,6 +25,32 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
+* The Jenkins Kubernetes plugin installed
+
+It first should be noted that the main reason to use the global pod template definition is to migrate a huge corpus of
```
Comment (Vlatombe) [inline README.md]:
```suggestion
It should be noted that the main reason to use the global pod template definition is to migrate a huge corpus of 
```

File: README.md [hunk 7]
```diff
@@ -25,6 +25,32 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
+Fill in the Kubernetes plugin configuration. In order to do that, you will open the Jenkins UI and navigate to 
+“Manage Jenkins -> Configure System -> Cloud -> Kubernetes” and enter in the ‘Kubernetes URL’ and ‘Jenkins URL’ 
+appropriately, this is unless Jenkins is running in Kubernetes in which case the defaults work. To test this connection is successful you can use the **Test Connection** button to ensure there is 
+adequate communication from Jenkins to the Kubernetes cluster, as seen below
+
+![image](Screen%20Shot%202020-02-20%20at%202.00.28%20PM.png)
+
+In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
+spin up the agent pod. We do not recommend overriding the `jnlp` container except under unusual circumstances. 
+for your agent, you can use the default Jenkins agent image available on the official Docker hub repository. In the
```
Comment (Vlatombe) [inline README.md]:
```suggestion
for your agent, you can use the default Jenkins agent image available in [Docker Hub](https://hub.docker.com). In the 
```

**Comment by markyjackson-taulia** [comment]
@Vlatombe Are we ok to merge this?

File: README.md [hunk 8]
```diff
@@ -25,6 +25,32 @@ Tested with [`jenkins/jnlp-slave`](https://hub.docker.com/r/jenkins/jnlp-slave),
...
+
+In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
+spin up the agent pod. We do not recommend overriding the `jnlp` container except under unusual circumstances. 
+for your agent, you can use the default Jenkins agent image available in [Docker Hub](https://hub.docker.com). In the 
+‘Kubernetes Pod Template’ section you need to specify the following (the rest of the configuration is up to you):
+Kubernetes Pod Template Name - can be any and will be shown as a prefix for unique generated agent’ names, which will 
+be run automatically during builds
+Docker image - the docker image name that will be used as a reference to spin up a new Jenkins agent, as seen below
+
+![image](Screen%20Shot%202020-02-20%20at%202.01.48%20PM.png)
```
Comment (jglick) [inline README.md]: Can we rename these images please, so that the filenames are descriptive and do not use dangerous characters?

Comment (Vlatombe) [reply README.md]: Already done.

**Pull request merged**